api/v1alpha1/driver_types.go

```diff
@@ -0,0 +1,268 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type LogSpec struct {
+	// Log level for driver pods,
+	// Supported values from 0 to 5. 0 for general useful logs (the default), 5 for trace level verbosity.
+	// Default to 0
+	LogLevel   int `json:"logLevel,omitempty"`
```

Comment:
We don't allow overwriting for specific fields inside the logging section, we do allow overwriting to the entire section as a whole - this means we don't need to use a pointer. 
Regarding 0 as a valid value: it is ok because it is also the default value

Reply:
if i set it to `0` as its default value it won't slow up in the json response, right? As a user i set it to `0` but when i do get operation i will not see it which can cause confusion

Reply:
As discussed offline. Omit empty is a marker for Go's JSON encoder. It has nothing to do with the CR definition. If a user will set the value to 0 on an instance of the CR, the value will be preserved